tests/test_compile_error.rs

```diff
@@ -1,6 +1,19 @@
 #[rustversion::stable]
 #[test]
 fn test_compile_errors() {
+    // stable - require all tests to pass
+    _test_compile_errors()
+}
+
+#[rustversion::nightly]
+#[test]
+fn test_compile_errors() {
+    // nightly - don't care if test output is potentially wrong, to avoid churn in PyO3's CI thanks
+    // to diagnostics changing on nightly.
+    let _ = std::panic::catch_unwind(_test_compile_errors);
```

Comment:
How about displaying some warnings?

Reply:
`trybuild` actually already prints a lot of warnings if the output is wrong, this just catches the `panic!` when trybuild attempts to fail the test. So I don't think we need to print anything else.